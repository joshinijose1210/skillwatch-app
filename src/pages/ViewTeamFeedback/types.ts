import { Feedback } from '@common';

export interface ViewTeamFeedbackState {
    name: string;
    reviewCycle: string;
    employeeId: number;
    reviewCycleId: number;
}

export interface FeedbackWithAppreciationInfo extends Feedback {
    appreciationFromFirstName?: string;
    appreciationToFirstName?: string;
    appreciationFromEmployeeId?: string;
    appreciationFromLastName?: string;
    appreciation?: string;
}

export interface FeedbackData {
    positiveFeedbackCount: number;
    improvementFeedbackCount: number;
    appreciationFeedbackCount: number;
    totalFeedbacks: number;
    feedbacks?: FeedbackWithAppreciationInfo[];
}
