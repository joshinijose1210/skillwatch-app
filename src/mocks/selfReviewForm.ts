const BASE_URL = process.env.API_URL;
const CURRENT_DAY = new Date().setHours(0, 0, 0);
const ONE_DAY = 86400000;
export const selfReviewFormMockData = {
    kpi: {
        url: `${BASE_URL}/api/kpi/fetch-by-employee-id?reviewToId=72&organisationId=1&reviewTypeId=1`,
        data: [
            {
                organisationId: 1,
                id: 2,
                title: 'Communicate when tasks are taking longer than usual',
                description: '<p>this is a test feedback response</p>',
                status: true,
                versionNumber: 1
            },
            {
                organisationId: 1,
                id: 4,
                title: 'Ask when tasks are taking longer than usual 2',
                description: '<p>this is a test feedback response 2</p>',
                status: true,
                versionNumber: 1
            },
            {
                organisationId: 1,
                id: 3,
                title: 'Tell when tasks are taking longer than usual 2',
                description: '<p>this is a test feedback response 2</p>',
                status: true,
                versionNumber: 2
            }
        ]
    },
    selfReview: {
        postUrl: 'https://dummy-api.example.com/api/self-review/',

        url: `https://dummy-api.example.com/api/self-review/?reviewTypeId=1&reviewCycleId=1&organisationId=1&reviewToId=72&reviewFromId=72`,
        data: [
            {
                reviewTypeId: 1,
                reviewDetailsId: 2,
                reviewCycleId: 1,
                reviewToId: 72,
                reviewToEmployeeId: 'SR021',
                reviewFromId: 72,
                reviewFromEmployeeId: 'SR021',
                draft: false,
                published: false,
                reviewData: [
                    {
                        reviewId: 1,
                        id: 2,
                        kpiTitle: 'Communicate when tasks are taking longer than usual',
                        kpiDescription: '<p>this is a test feedback</p>',
                        review: '<p><span>This is test response, we are checking response form of self review.</span></p>',
                        rating: -1
                    },
                    {
                        reviewId: 1,
                        id: 4,
                        kpiTitle: 'Ask when tasks are taking longer than usual',
                        kpiDescription: '<p>this is a test feedback</p>',
                        review: '<p><span>This is test response, we are checking response form of self review.</span></p>',
                        rating: 3
                    },
                    {
                        reviewId: 1,
                        id: 3,
                        kpiTitle: 'Tell when tasks are taking longer than usual',
                        kpiDescription: '<p>this is a test feedback</p>',
                        review: '<p><span>This is test response, we are checking response form of self review.</span></p>',
                        rating: 2
                    }
                ]
            }
        ]
    },
    mockState: {
        organisationId: 1,
        reviewCycleId: 1,
        startDate: CURRENT_DAY - ONE_DAY,
        endDate: CURRENT_DAY + ONE_DAY * 30,
        publish: true,
        selfReviewStartDate: CURRENT_DAY + ONE_DAY,
        selfReviewEndDate: CURRENT_DAY + ONE_DAY * 5,
        managerReviewStartDate: CURRENT_DAY + ONE_DAY,
        managerReviewEndDate: CURRENT_DAY + ONE_DAY * 5,
        selfReviewDraft: false,
        selfReviewPublish: false,
        selfReviewDate: CURRENT_DAY,
        selfAverageRating: 5,
        firstManagerId: 54,
        firstManagerEmployeeId: 'SR0039',
        firstManagerFirstName: 'Aamir',
        firstManagerLastName: 'Islam',
        firstManagerReviewDraft: false,
        firstManagerReviewPublish: false,
        firstManagerAverageRating: -1,
        secondManagerReviewDraft: false,
        secondManagerReviewPublish: false,
        secondManagerAverageRating: -1,
        checkInWithManagerStartDate: CURRENT_DAY + ONE_DAY * 6,
        checkInWithManagerEndDate: CURRENT_DAY + ONE_DAY * 9,
        checkInWithManagerDraft: false,
        checkInWithManagerPublish: false,
        checkInWithManagerAverageRating: -1,
        isOrWasManager: true
        // "empDetails": [
        //   {
        //     "id": 95,
        //     "employeeId": "SR0056",
        //     "firstName": "Rahil",
        //     "lastName": "Khan",
        //     "firstManagerId": 54,
        //     "secondManagerId": 44,
        //     "selfReviewDraft": false,
        //     "selfReviewPublish": false,
        //     "firstManagerReviewDraft": false,
        //     "firstManagerReviewPublish": false,
        //     "secondManagerReviewDraft": false,
        //     "secondManagerReviewPublish": false,
        //     "checkInWithManagerDraft": false,
        //     "checkInWithManagerPublish": false
        //   },
        // ]
    },
    editMockState: {
        reviewCycleId: 1,
        startDate: CURRENT_DAY - ONE_DAY,
        endDate: CURRENT_DAY + ONE_DAY * 30,
        selfReviewStartDate: CURRENT_DAY + ONE_DAY,
        selfReviewEndDate: CURRENT_DAY + ONE_DAY * 5,
        draft: true,
        publish: false,
        updatedAt: CURRENT_DAY,
        averageRating: 2.3
    },
    viewTeamFeedbackMockData: {
        feedback: {
            url: 'https://dummy-api.example.com/api/dashboard/employee-feedback/?id=72&organisationId=1&feedbackTypeId=1&reviewCycleId=1',
            data: {
                positiveFeedbackCount: 2,
                improvementFeedbackCount: 1,
                appreciationFeedbackCount: 3,
                totalFeedbacks: 2,
                feedbacks: [
                    {
                        feedbackType: 'Positive',
                        feedback: '<p>Test text</p>',
                        feedbackFromId: 73,
                        feedbackFromEmployeeId: 'SR0022',
                        feedbackFromFirstName: 'A',
                        feedbackFromLastName: 'G',
                        feedbackFromRoleName: 'Manager',
                        submitDate: 1689754057181,
                        isDraft: false
                    },
                    {
                        feedbackType: 'Positive',
                        feedback:
                            '<p><span style="color: rgb(96, 120, 144);">Positive feedback is a way to recognize and appreciate a job well done. When giving positive feedback:</span></p><ul><li>Be specific: Identify what the person did well and why it was important.</li><li>Use "I" statements: Focus on your own experience and reactions, rather than making judgments or assumptions about the other person.</li></ul><p><br></p>',
                        feedbackFromId: 74,
                        feedbackFromEmployeeId: 'SR0050',
                        feedbackFromFirstName: 'M',
                        feedbackFromLastName: 'A',
                        feedbackFromRoleName: 'Human Resource',
                        submitDate: 1688735276632,
                        isDraft: false
                    }
                ]
            }
        }
    }
};
