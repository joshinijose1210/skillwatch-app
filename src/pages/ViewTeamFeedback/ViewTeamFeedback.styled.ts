import { ParentStyledTabs } from '@common';
import { Text } from '@medly-components/core';
import styled from 'styled-components';

export const FeedbackWrapper = styled.div`
    border-bottom: 2px solid ${({ theme }) => theme.colors.grey[200]};
    padding: 1rem 1rem 2rem;
    display: flex;
    gap: 2px;
    p {
        margin: 0;
    }
    .team-feedback {
        overflow-wrap: break-word;
        width: 100%;
    }
    &:last-child {
        border-bottom: none;
    }
`;

export const StyledTabs = styled(ParentStyledTabs)`
    #feedback-tabs-list {
        overflow: hidden;
    }
    #feedback-tabs-panel-positive,
    #feedback-tabs-panel-constructive,
    #feedback-tabs-panel-improvement,
    #feedback-tabs-panel-appreciation {
        margin: 0;
        padding: 1rem;
        border: 2px solid ${({ theme }) => theme.colors.blue[200]};
        border-radius: 5px;
        border-top-left-radius: 0;
        width: 98%;
    }

    button[aria-selected='false'] div div span:nth-of-type(2) {
        color: #666;
        background-color: ${({ theme }) => theme.colors.grey[200]};
    }

    button[aria-selected='false']:hover div div span:nth-of-type(2) {
        background-color: ${({ theme }) => theme.colors.grey[200]};
    }

    button[aria-selected='false'] svg path {
        fill: #666;
    }
`;

export const TabContentWrapper = styled.div`
    border: 2px solid aliceblue;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
    width: 100%;
`;

export const RightColumn = styled.div``;

export const Feedback = styled.div`
    margin-bottom: 1rem;
    font-size: 1.5rem;
    line-height: 1.8rem;
    letter-spacing: 0;
`;

export const StyledDate = styled(Text)`
    line-height: 0.5rem;
    color: ${({ theme }) => theme.customColors.fullGrey};
    opacity: 0.5;
    font-weight: 700;
    margin-top: 5px;
    padding-left: 1.2rem;
`;
