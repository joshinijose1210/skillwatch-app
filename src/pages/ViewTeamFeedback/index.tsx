export { ViewTeamFeedback as default } from './ViewTeamFeedback';
