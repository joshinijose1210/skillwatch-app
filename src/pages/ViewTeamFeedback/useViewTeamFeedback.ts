import { useAppSelector } from '@slice';
import { useGetAppreciationsMutation } from '@slice/services/dashboardAppreciation';
import { useCallback, useEffect, useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { FeedbackData, ViewTeamFeedbackState } from './types';

export const useViewTeamFeedback = () => {
    const userDetails = useAppSelector(state => state.user);
    const [tagId, setTagId] = useState(1);
    const [feedbackData, setFeedbackData] = useState<FeedbackData>({
        positiveFeedbackCount: 0,
        improvementFeedbackCount: 0,
        appreciationFeedbackCount: 0,
        totalFeedbacks: 0,
        feedbacks: []
    });

    const navigateTo = useNavigate(),
        state = useLocation().state as ViewTeamFeedbackState;
    const [getFeedback, { data, isSuccess, isLoading }] = useGetAppreciationsMutation();

    const handleBackClick = () => {
        navigateTo(-1);
    };

    useEffect(() => {
        if (isSuccess) {
            setFeedbackData(data);
        }
    }, [data, isSuccess]);

    const onChangeHandle = useCallback((id: string) => {
        setFeedbackData(prev => ({
            ...prev,
            feedbacks: []
        }));
        if (id === 'positive') {
            setTagId(1);
        } else if (id === 'improvement') {
            setTagId(2);
        } else {
            setTagId(3);
        }
    }, []);

    useEffect(() => {
        getFeedback({
            path: '',
            params: [
                { name: 'id', value: state && state.employeeId },
                { name: 'organisationId', value: userDetails.organisationId },
                { name: 'feedbackTypeId', value: tagId },
                { name: 'reviewCycleId', value: state.reviewCycleId }
            ]
        });
    }, [state, getFeedback, tagId, userDetails.organisationId]);

    return {
        data,
        feedbackData,
        isLoading,
        state,
        handleBackClick,
        onChangeHandle
    };
};
