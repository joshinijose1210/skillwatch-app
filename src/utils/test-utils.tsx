import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
