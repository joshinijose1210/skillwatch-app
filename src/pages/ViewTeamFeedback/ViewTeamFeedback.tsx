import { FromText } from '@common';
import { Loader, PageContent } from '@components';
import Grid from '@components/layout/Grid';
import ListHeader from '@components/reusableComponents/ListHeader';
import { Tabs, Text } from '@medly-components/core';
import { format } from 'date-fns';
import DOMPurify from 'dompurify';
import { FC } from 'react';
import { useViewTeamFeedback } from './useViewTeamFeedback';
import { Feedback, FeedbackWrapper, StyledDate, StyledTabs } from './ViewTeamFeedback.styled';

export const ViewTeamFeedback: FC = () => {
    const { isLoading, state, handleBackClick, onChangeHandle, feedbackData } = useViewTeamFeedback();
    return (
        <PageContent>
            <ListHeader
                title={
                    <>
                        Team&apos;s Feedback{' '}
                        <Text style={{ fontSize: '14px' }} textVariant="body1" textWeight="Medium">
                            ({state.reviewCycle})
                        </Text>
                    </>
                }
                actionButtonVariant="flat"
                actionButtonLabel={'Go Back'}
                actionButtonClick={handleBackClick}
                moduleTitle="Manager Review"
            />

            <StyledTabs data-testid="tabs" id="feedback-tabs" variant="outlined" onChange={onChangeHandle}>
                <Tabs.Tab id="positive" label="Positive" count={feedbackData.positiveFeedbackCount}>
                    {isLoading ? (
                        <Loader />
                    ) : feedbackData.totalFeedbacks > 0 ? (
                        feedbackData.feedbacks &&
                        feedbackData.feedbacks.map((feedback, index: number) => (
                            <FeedbackWrapper key={index}>
                                <Text textVariant="body2" textWeight="Medium">
                                    {index + 1}.
                                </Text>
                                <Grid column direction="column-direction" customClasses="team-feedback">
                                    <Feedback dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(feedback.feedback) }} />
                                    <FromText textVariant="h5">
                                        &#8211;{' '}
                                        {feedback.isExternalFeedback
                                            ? feedback.externalFeedbackFromEmailId
                                            : `${feedback.feedbackFromFirstName} ${feedback.feedbackFromLastName} (
                                                ${feedback.feedbackFromEmployeeId})`}
                                    </FromText>
                                    <StyledDate textVariant="h5">{format(new Date(feedback.submitDate), 'dd/MM/yyyy')}</StyledDate>
                                </Grid>
                            </FeedbackWrapper>
                        ))
                    ) : (
                        <Text textVariant="body2" textAlign="center">
                            No feedbacks.
                        </Text>
                    )}
                </Tabs.Tab>
                <Tabs.Tab id="improvement" label="Improvement" count={feedbackData.improvementFeedbackCount}>
                    {isLoading ? (
                        <Loader />
                    ) : feedbackData.totalFeedbacks > 0 ? (
                        feedbackData.feedbacks &&
                        feedbackData.feedbacks.map((feedback, index: number) => (
                            <FeedbackWrapper key={index}>
                                <Text textVariant="body2" textWeight="Medium">
                                    {index + 1}.
                                </Text>
                                <Grid column direction="column-direction" customClasses="team-feedback">
                                    <Feedback dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(feedback.feedback) }} />
                                    <FromText textVariant="h5">
                                        &#8211;{' '}
                                        {feedback.isExternalFeedback
                                            ? feedback.externalFeedbackFromEmailId
                                            : `${feedback.feedbackFromFirstName} ${feedback.feedbackFromLastName} (
                                                ${feedback.feedbackFromEmployeeId})`}
                                    </FromText>
                                    <StyledDate textVariant="h5">{format(new Date(feedback.submitDate), 'dd/MM/yyyy')}</StyledDate>
                                </Grid>
                            </FeedbackWrapper>
                        ))
                    ) : (
                        <Text textVariant="body2" textAlign="center">
                            No feedbacks.
                        </Text>
                    )}
                </Tabs.Tab>
                <Tabs.Tab id="appreciation" label="Appreciation" count={feedbackData.appreciationFeedbackCount}>
                    {isLoading ? (
                        <Loader />
                    ) : feedbackData.totalFeedbacks > 0 ? (
                        feedbackData.feedbacks &&
                        feedbackData.feedbacks.map((feedback, index: number) => (
                            <FeedbackWrapper key={index}>
                                <Text textVariant="body2" textWeight="Medium">
                                    {index + 1}.
                                </Text>
                                <Grid column direction="column-direction" customClasses="team-feedback">
                                    <Feedback dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(feedback?.appreciation || '') }} />
                                    <FromText textVariant="h5">
                                        &#8211;{' '}
                                        {feedback.isExternalFeedback
                                            ? feedback.externalFeedbackFromEmailId
                                            : `${feedback.appreciationFromFirstName} ${feedback.appreciationFromLastName} (
                                                ${feedback.appreciationFromEmployeeId})`}
                                    </FromText>
                                    <StyledDate textVariant="h5">{format(new Date(feedback.submitDate), 'dd/MM/yyyy')}</StyledDate>
                                </Grid>
                            </FeedbackWrapper>
                        ))
                    ) : (
                        <Text textVariant="body2" textAlign="center">
                            No feedbacks.
                        </Text>
                    )}
                </Tabs.Tab>
            </StyledTabs>
        </PageContent>
    );
};
