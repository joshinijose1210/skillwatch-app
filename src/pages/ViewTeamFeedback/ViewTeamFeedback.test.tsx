import { RTKRender, fireEvent, screen } from '@utils/test-utils';
import { ViewTeamFeedback } from './ViewTeamFeedback';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { ManagerReviewMockData } from '@mocks/managerReview';
import * as reactRouterDom from 'react-router-dom';
import { format } from 'date-fns';
import { selfReviewFormMockData } from '@mocks/selfReviewForm';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn()
}));

const currentData = new Date('2024-10-18T00:00:00Z');
const reviewCycle = `${format(currentData, 'do MMM yyyy')} - ${format(currentData.getDate() + 5, 'do MMM yyyy')}`;

beforeAll(() => {
    const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');

    useLocationMock.mockReturnValue({
        ...ManagerReviewMockData.LocationMockData,
        state: { name: 'Back to self review', reviewCycle, employeeId: 72, reviewCycleId: 1 }
    });
});

describe('ViewTeamFeedback component', () => {
    it('should take snapshot', () => {
        const { container } = RTKRender(<ViewTeamFeedback />);
        expect(container).toMatchSnapshot();
    });

    it('should render properly', async () => {
        RTKRender(<ViewTeamFeedback />);
        expect(screen.getByText(`Team's Feedback`)).toBeInTheDocument();
        await screen.findByText(`No feedbacks.`);
    });

    it('should go back properly', () => {
        RTKRender(<ViewTeamFeedback />);
        expect(screen.getByText(`Team's Feedback`)).toBeInTheDocument();
        fireEvent.click(screen.getByText('Go Back'));
        expect(mockNavigate).toBeCalledWith(-1);
    });

    it('should tab change properly', async () => {
        const { url, data } = selfReviewFormMockData.viewTeamFeedbackMockData.feedback;
        mock.onGet(url).reply(200, data);
        RTKRender(<ViewTeamFeedback />);
        expect(screen.getByText(`Team's Feedback`)).toBeInTheDocument();
        const Improvement = screen.getByText('Improvement');
        const Appreciation = screen.getByText('Appreciation');
        const positive = screen.getByText('Positive');
        fireEvent.click(Appreciation);
        fireEvent.click(Improvement);
        fireEvent.click(positive);
        await screen.findByText('Test text');
        expect(positive).toHaveStyle({ color: 'rgb(56, 114, 210)' });
    });
});
