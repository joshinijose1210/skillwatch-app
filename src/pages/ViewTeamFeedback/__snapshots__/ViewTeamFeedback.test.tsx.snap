// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTeamFeedback component should take snapshot 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Team's Feedback
           
          <span
            class="sc-1ntgl3l-0 dUPZoS"
            style="font-size: 14px;"
            textalign="initial"
            textcolor="inherit"
            textvariant="body1"
            textweight="Medium"
          >
            (
            18th Oct 2024 - 1st Jan 1970
            )
          </span>
        </h3>
      </div>
      <button
        class="sc-1k2druj-0 yZQJV sc-w8e7z6-1 lnjfja"
        edges="square"
        type="button"
        variant="flat"
      >
        <span
          class="sc-1ntgl3l-0 eOsHHD"
          textalign="initial"
          textcolor="inherit"
          textvariant="button1"
        >
          Go Back
        </span>
      </button>
    </div>
    <div
      class="sc-13bppbd-0 sc-tkiih6-28 sc-1tw5u2q-1 dErHOx elfdNn"
      data-testid="tabs"
      id="feedback-tabs"
    >
      <div
        class="sc-12mpb33-0 dYQjJT"
        id="feedback-tabs-list"
        tabsize="S"
        variant="outlined"
      >
        <button
          aria-controls="panel-positive"
          aria-selected="true"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 hLhToq"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="positive"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="0"
          tabsize="S"
          totaltabs="3"
          type="button"
          variant="outlined"
        >
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 ieSRdN"
                id="positive-label"
                tabsize="S"
                textalign="initial"
                textcolor="inherit"
              >
                Positive
              </span>
              <span
                class="sc-1kcg570-0 bVzkba"
                id="positive-label"
                tabsize="S"
              >
                0
              </span>
            </div>
          </div>
        </button>
        <button
          aria-controls="panel-improvement"
          aria-selected="false"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 hzTrVV"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="improvement"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="-1"
          tabsize="S"
          totaltabs="3"
          type="button"
          variant="outlined"
        >
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 ieSRdN"
                id="improvement-label"
                tabsize="S"
                textalign="initial"
                textcolor="inherit"
              >
                Improvement
              </span>
              <span
                class="sc-1kcg570-0 bVzkba"
                id="improvement-label"
                tabsize="S"
              >
                0
              </span>
            </div>
          </div>
        </button>
        <button
          aria-controls="panel-appreciation"
          aria-selected="false"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 hzTrVV"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="appreciation"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="-1"
          tabsize="S"
          totaltabs="3"
          type="button"
          variant="outlined"
        >
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 ieSRdN"
                id="appreciation-label"
                tabsize="S"
                textalign="initial"
                textcolor="inherit"
              >
                Appreciation
              </span>
              <span
                class="sc-1kcg570-0 bVzkba"
                id="appreciation-label"
                tabsize="S"
              >
                0
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        aria-labelledby="positive"
        class="sc-x0gfjw-0 bLZBSc"
        id="feedback-tabs-panel-positive"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="sc-mbkfkf-0 KnoHL"
        >
          <main
            class="sc-1ehavxz-0 dDyhBJ"
          >
            <div
              class="sc-6d0kjn-0 dkGZbP"
            >
              <svg
                class="sc-433e5x-0 dVjAFq"
                height="1em"
                viewBox="0 0 100 100"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M31.6 3.5C5.9 13.6-6.6 42.7 3.5 68.4c10.1 25.7 39.2 38.3 64.9 28.1l-3.1-7.9C44 97 19.9 86.6 11.5 65.3c-8.4-21.3 2-45.4 23.3-53.8l-3.2-8z"
                >
                  <animatetransform
                    attributeName="transform"
                    attributeType="XML"
                    dur="2s"
                    from="0 50 50"
                    repeatCount="indefinite"
                    to="360 50 50"
                    type="rotate"
                  />
                </path>
                <path
                  d="M42.3 39.6c5.7-4.3 13.9-3.1 18.1 2.7 4.3 5.7 3.1 13.9-2.7 18.1l4.1 5.5c8.8-6.5 10.6-19 4.1-27.7-6.5-8.8-19-10.6-27.7-4.1l4.1 5.5z"
                >
                  <animatetransform
                    attributeName="transform"
                    attributeType="XML"
                    dur="1s"
                    from="0 50 50"
                    repeatCount="indefinite"
                    to="-360 50 50"
                    type="rotate"
                  />
                </path>
                <path
                  d="M82 35.7C74.1 18 53.4 10.1 35.7 18S10.1 46.6 18 64.3l7.6-3.4c-6-13.5 0-29.3 13.5-35.3s29.3 0 35.3 13.5l7.6-3.4z"
                >
                  <animatetransform
                    attributeName="transform"
                    attributeType="XML"
                    dur="2s"
                    from="0 50 50"
                    repeatCount="indefinite"
                    to="360 50 50"
                    type="rotate"
                  />
                </path>
              </svg>
            </div>
          </main>
        </div>
      </div>
    </div>
  </main>
</div>
`;
